lib/db.c

```diff
@@ -6075,7 +6075,9 @@ grn_table_setoperation_or(grn_ctx *ctx,
                                               key_src_size,
                                               &value_dest,
                                               &added);
-      if (id_dest != GRN_ID_NIL) {
+      if (id_dest == GRN_ID_NIL) {
+        break;
```

Comment:
Could you also check `ctx->rc`? I'm not sure we can break for all "failed to add" cases.

```c
if (id_dest == GRN_ID_NIL) {
  if (ctx->rc == GRN_SUCCESS) {
    continue;
  } else{
    break;
  }
}
```

BTW, do we need the same logic for `!dat->have_subrec` case too?

Reply:
I see. I also add a check of `ctx->rc`.

> BTW, do we need the same logic for !data->have_subrec case too?
Sure.
We need this logic for `!data->have_subrec` case. I added at d4a013b40bdefdbc4eab25f49f2333ae71da0aa5.